app.go

```diff
@@ -622,3 +621,53 @@ func (a *Application) ReadOnlyState() appstate.State {
 		a.GetValidatorSet,
 	)
 }
+
+func (a *Application) ReplayApplication(blockNumber uint64, blockstore store.BlockStore) (*Application, int64, error) {
```

Comment:
So now that it's here instead of a separate package you should be able to eliminate most of the refactoring in this PR, the tx handlers don't need to move, the middleware doesn't need to move, the configs don't need to move. Probably easier to copy the relevant bits out to a fresh PR.